accumulation/analyzer.go

```diff
@@ -170,14 +170,11 @@ func errorsToDiagnostics(errs []error) []analysis.Diagnostic {
 // buildDiagnostics takes a list of FullTriggers, which are assumed to already have been checked
 // and are known to fail, and returns a slice of the appropriate diagnostics for all the triggers.
 func buildDiagnostics(pass *analysis.Pass, triggers []annotation.FullTrigger) []analysis.Diagnostic {
-	var errors []analysis.Diagnostic
+	conflicts := inference.ConflictList{NoGrouping: true}
```

Comment:
Right, as we discussed, separating the diagnostics engine into its own package would make the separation a lot better!

Reply:
We don't want grouping since no-infer mode is mostly used for tests and we actually want separate errors for better testability, right? If so, can we add a comment here for future readers?

Reply:
Correct. Comment added.